### PR TITLE
fix reading MIDC files with mismatching header/data columns

### PR DESCRIPTION
**Pull request by alorenzo175**
- [x] Closes issue #xxxx
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [x] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

Requesting data from the MIDC API sometimes (e.g. for the UTPASRL site on 2019-11-15) returns a csv file where the number of header columns and data columns do no match. This leads pandas to create a multi-index with the extra columns (with the header applied to the rightmost columns) with the remaining columns mislabelled as a result. This PR first tries to parses the headers from the file and then only parses the first ``len(headers)`` columns of the data.

### CODE REVIEW

File: pvlib/iotools/midc.py [hunk 1]
```diff
@@ -148,21 +152,23 @@ def format_index_raw(data):
...
+        Filename, url, or file-like object of data to read.
     variable_map: dictionary
         Dictionary for mapping MIDC field names to pvlib names. Used to rename
         the columns of the resulting DataFrame. Does not map names by default.
         See Notes for an example.
     raw_data: boolean
         Set to true to use format_index_raw to correctly format the date/time
         columns of MIDC raw data files.
+    kwargs : dict
+       Additonal keyword arguments to pass to `pandas.read_csv`
```
Comment (wholmgren) [inline pvlib/iotools/midc.py]:
```suggestion
       Additional keyword arguments to pass to `pandas.read_csv`
```

File: pvlib/iotools/midc.py [hunk 2]
```diff
@@ -236,4 +247,14 @@ def read_midc_raw_data_from_nrel(site, start, end, variable_map={}):
             'end': end.strftime('%Y%m%d')}
     endpoint = 'https://midcdmz.nrel.gov/apps/data_api.pl?'
     url = endpoint + '&'.join(['{}={}'.format(k, v) for k, v in args.items()])
-    return read_midc(url, variable_map=variable_map, raw_data=True)
+    # number of header columns and data columns do not always match,
+    # so first parse the header to determine the number of data columns
+    # to parse
+    csv_request = requests.get(url)
```
Comment (cwhanse) [inline pvlib/iotools/midc.py]: Include `timeout` here? We did for `get_psm3` but I pushed for that addition and got the impression that `timeout` wasn't seen as necessary by all.

Comment (alorenzo175) [reply pvlib/iotools/midc.py]: Yes, something I overlooked and definitely a good idea to include.

**Comment by cwhanse** [comment]
@wholmgren OK with me to merge

**Pull request merged**